src/movies.js

```diff
@@ -1,22 +1,61 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
-
+function getAllDirectors(moviesArray) {
+    let film=moviesArray.map(function(cinema){
+       return cinema.director
+    })
+return film
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
-
+function howManyMovies(moviesArray) {
+    let countDown=moviesArray.filter((cinema2) => cinema2.director==="Steven Spielberg"&& cinema2.genre.includes("Drama"))
+    return countDown.length
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+if (moviesArray.length === 0) {
```

Comment:
Everything here is ok but you deleted this line:

`function scoresAverage(moviesArray) {`